nodes/test/test_multi_agent.py

```
@@ -0,0 +1,890 @@
+            AgentDefinition.from_dict({'name': 'bot', 'evil_key': 'payload'})
+
+    def test_from_dict_rejects_non_dict(self):
+        with pytest.raises(ValueError, match='must be a dict'):
+            AgentDefinition.from_dict('not a dict')
+
+    def test_empty_name_raises(self):
+        with pytest.raises(ValueError, match='non-empty string'):
+            AgentDefinition(name='')
+
+    def test_name_must_be_string(self):
+        with pytest.raises(ValueError):
```

Ruff (0.15.7)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)